staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields.go

```diff
@@ -110,7 +110,30 @@ func EncodeObjectManagedFields(obj runtime.Object, managed ManagedInterface) err
 func decodeManagedFields(encodedManagedFields []metav1.ManagedFieldsEntry) (managed managedStruct, err error) {
 	managed.fields = make(fieldpath.ManagedFields, len(encodedManagedFields))
 	managed.times = make(map[string]*metav1.Time, len(encodedManagedFields))
-	for _, encodedVersionedSet := range encodedManagedFields {
+
+	// If we have exactly one managedFields with no manager and no
+	// version, keep an empty manager to get a reset.
```

Comment:
What circumstances produce an entry like this? I'm not familiar enough with the data flow to know where this is coming from, but isn't an empty manager and apiVersion invalid?